### PR TITLE
[TM-523] Bump morley and other deps

### PR DESCRIPTION
**Pull request by rvem**
Problem: We want to check whether this repo can be used with the latest
morley and new GHC version.

Solution: Bump dependencies and make required changes in the code.

## Related issue(s)

https://issues.serokell.io/issue/TM-523

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock
  - [x] I updated [changelog](../tree/master/ChangeLog.md) unless I am sure my changes are
        not essential.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Comment by pasqu4le** [comment]
The changelog should be updated, also while the gas cost should only be updated after the next release I think the rest of the README should be updated now (particularly the instructions to deploy the contract, which are based on dead testnets).

File: haskell/src/Stablecoin/Client/Cleveland/IO.hs [hunk 1]
```diff
@@ -53,13 +55,13 @@ callStablecoinClient conf args = toText <$> do
...
-  <> ["--client-path", tzPath]
-  <> encodeMaybeOption "--data-dir" dataDir
-  <> replicate (fromIntegral verb) "-V"
-  <> encodeMaybeOption "--secret-key" key
+morleyClientOpts :: MorleyClientEnv -> [String]
+morleyClientOpts (MorleyClientEnv tcEnv _ mbSk _) =
+  ["--endpoint", showBaseUrl $ tceEndpointUrl tcEnv]
+  <> ["--client-path", tceTezosClientPath tcEnv]
+  <> encodeMaybeOption "--data-dir" (tceMbTezosClientDataDir tcEnv)
+  -- <> replicate (fromIntegral verb) "-V"
```
Comment (pasqu4le) [inline haskell/src/Stablecoin/Client/Cleveland/IO.hs:63]: is this a leftover?

File: haskell/src/Stablecoin/Client/Contract.hs [hunk 2]
```diff
@@ -47,12 +47,12 @@ mkInitialStorage :: InitialStorageData Address -> Storage
 mkInitialStorage InitialStorageData {..} =
   Storage
     { sDefaultExpiry = isdDefaultExpiry
-    , sLedger =  mempty
+    , sLedger =  mkBigMap mempty
```
Comment (pasqu4le) [inline haskell/src/Stablecoin/Client/Contract.hs:50]:
```suggestion
    , sLedger = mkBigMap mempty
```
:detective:

**Review by pasqu4le — COMMENTED** [review]
Minor comments, but overall LGTM :ok_hand:

File: haskell/nettest/FA1_2Comparison.hs [hunk 3]
```diff
@@ -55,7 +56,7 @@ fa1_2ComparisonScenario = uncapsNettest $ do
   cmrFA1_2Address <- nettestOriginateContractMetadataContract metadata
   let fa1_2Storage = SFA1_2.Storage
         { sDefaultExpiry = 1000
-        , sLedger = BigMap balances
+        , sLedger = BigMap Nothing balances
```
[the remark this answers is not in the document]
Comment (rvem) [reply haskell/nettest/FA1_2Comparison.hs:59]: At the moment, `mbBigMap` only accepts a list of pair, while `balances` is already a `Map`

[the remark this answers is not in the document]
Comment (rvem) [reply haskell/nettest/FA1_2Comparison.hs:59]: nevermind

**Review by pasqu4le — APPROVED** [review]

**Pull request merged**